KIPs/kip-draft_kgp6.md

```diff
@@ -0,0 +1,714 @@
+     * @param _newbieAddress is the address of the newbie
+     * @param _amount is the fund to be allocated to the newbie
+     */
+    function registerNewbie(
+        address _newbieAddress,
+        uint256 _amount
+    ) public onlyOwner onlyAtStatus(Status.Initialized) {
+        require(
+            !newbieExists(_newbieAddress),
+            "Newbie address is already registered"
+        );
+        require(_amount != 0, "Amount cannot be set to 0");
+
+        Newbie memory newbie = Newbie(_newbieAddress, _amount);
+        newbies.push(newbie);
+
+        emit NewbieRegistered(_newbieAddress, _amount);
+    }
+
+    /**
+     * @dev remove the newbie details from the array
+     * @param _newbieAddress is the address of the newbie
+     */
+    function removeNewbie(
+        address _newbieAddress
+    ) public onlyOwner onlyAtStatus(Status.Initialized) {
+        uint256 newbieIndex = getNewbieIndex(_newbieAddress);
+        newbies[newbieIndex] = newbies[newbies.length - 1];
+        newbies.pop();
+
+        emit NewbieRemoved(_newbieAddress, newbies.length);
+    }
+
+    /**
+     * @dev retiredAddress can be a EOA or a contract address. To approve:
+     *      If the retiredAddress is a EOA, the msg.sender should be the EOA address
+     *      If the retiredAddress is a Contract, the msg.sender should be one of the contract `admin`.
+     *      It uses the getState() function in the retiredAddress contract to get the admin details.
+     * @param _retiredAddress is the address of the retired
+     */
+    function approve(
+        address _retiredAddress
+    ) public onlyAtStatus(Status.Registered) {
+        require(
+            retiredExists(_retiredAddress),
+            "retired needs to be registered before approval"
+        );
+
+        //Check whether the retired address is EOA or contract address
+        bool isContract = isContractAddr(_retiredAddress);
+        if (!isContract) {
+            //check whether the msg.sender is the retired if its a EOA
+            require(
+                msg.sender == _retiredAddress,
+                "retiredAddress is not the msg.sender"
+            );
+            _updateApprover(_retiredAddress, msg.sender);
+        } else {
+            (address[] memory adminList, ) = _getState(_retiredAddress);
+            require(adminList.length != 0, "admin list cannot be empty");
+
+            //check if the msg.sender is one of the admin of the retiredAddress contract
+            require(
+                _validateAdmin(msg.sender, adminList),
+                "msg.sender is not the admin"
+            );
+            _updateApprover(_retiredAddress, msg.sender);
+        }
+    }
+
+    /**
+     * @dev validate if the msg.sender is admin if the retiredAddress is a contract
+     * @param _approver is the msg.sender
+     * @return isAdmin is true if the msg.sender is one of the admin
+     */
+    function _validateAdmin(
```

Comment:
can we make this function as a view function?

Reply:
Done